src/parser/regexp.bnf

```diff
@@ -694,6 +703,8 @@ NonemptyClassRanges
 
   | ClassAtom DASH ClassAtom ClassRanges
     {
+      checkClassRange($1.value, $3.value);
```

Comment:
@silentroach, would you like to proceed with this one? Or I can merge this request, and do a follow up with fixing this part.